concordium-std/src/impls.rs

```diff
@@ -2083,16 +2083,25 @@ where
         }
     }
 
+    /// Provide clone of [`HasStateApi`] instance and new key prefix
+    /// for any container-like type wishing to store its data on blockchain
+    ///
+    /// Contract developers can use this method to implement their own
+    /// containers
```

Comment:
To make this clearer, can we add a reference to StateMap and StateSet?

Reply:
Added more detailed description on what method does and roughly how to use it. Please recheck.

Reply:
Looks good.

Reply:
Thx, resolving this thread then